File: Android5pmClasses/TypesOfAdaptes/app/src/main/java/com/cbitss/typesofadaptes/ArrayAdapterExample.java
package com.cbitss.typesofadaptes;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ListView;

import java.util.ArrayList;

public class ArrayAdapterExample extends AppCompatActivity {
ListView listView;
ArrayList<Item>  userlist = new ArrayList<>();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_array_adapter_example);

        listView = findViewById(R.id.listview_array_adtapter);

        userlist.add(new Item("Lion",R.drawable.arebic));
        userlist.add(new Item("Tiger",R.drawable.sample));
        userlist.add(new Item("Lion",R.drawable.arebic));
        userlist.add(new Item("Lion",R.drawable.arebic));
        userlist.add(new Item("Lion",R.drawable.arebic));
        userlist.add(new Item("Tiger",R.drawable.sample));
        userlist.add(new Item("Lion",R.drawable.arebic));
        userlist.add(new Item("Lion",R.drawable.arebic));

        CustomArrayAdapter customArrayAdapter = new CustomArrayAdapter(getApplicationContext(),R.layout.list_formate,userlist);
        listView.setAdapter(customArrayAdapter);




    }
}

File: Rahoul/TypesofAdapter/app/src/main/java/com/example/typesofadapter/ArrayAdapterExample.kt
package com.example.typesofadapter

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ListView
import android.widget.Toast

class ArrayAdapterExample : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_array_adapter_example)
        var listview = findViewById<ListView>(R.id.array_listve_view)
        var listviewdata : MutableList<Item> = ArrayList<Item>()

        listviewdata.add(Item("First",R.drawable.arebic))
        listviewdata.add(Item("Second",R.drawable.download))
        listviewdata.add(Item("Third",R.drawable.first))
        listviewdata.add(Item("First",R.drawable.arebic))
        listviewdata.add(Item("Second",R.drawable.download))
        listviewdata.add(Item("Third",R.drawable.first))
        listviewdata.add(Item("First",R.drawable.arebic))
        listviewdata.add(Item("Second",R.drawable.download))
        listviewdata.add(Item("Third",R.drawable.first))

        var adapter = CustomAdapter(applicationContext,listviewdata,R.layout.array_rows)

        listview.adapter = adapter
     listview.setOnItemClickListener { parent, view, position, id ->

         Toast.makeText(applicationContext,"clickec item at position ${listviewdata.get(position).name}",Toast.LENGTH_SHORT).show()
     }
    }
}